Foundation/NSJSONSerialization.swift

```diff
@@ -93,7 +93,74 @@ public class NSJSONSerialization : NSObject {
     /* Generate JSON data from a Foundation object. If the object will not produce valid JSON then an exception will be thrown. Setting the NSJSONWritingPrettyPrinted option will generate JSON with whitespace designed to make the output more readable. If that option is not set, the most compact possible JSON will be generated. If an error occurs, the error parameter will be set and the return value will be nil. The resulting data is a encoded in UTF-8.
      */
     public class func dataWithJSONObject(obj: AnyObject, options opt: NSJSONWritingOptions) throws -> NSData {
-        NSUnimplemented()
+        let newIndent = opt.contains(NSJSONWritingOptions.PrettyPrinted) ? "\t" : ""
+        let newLine = opt.contains(NSJSONWritingOptions.PrettyPrinted) ? "\n" : ""
+        let spacing = opt.contains(NSJSONWritingOptions.PrettyPrinted) ? " " : ""
+        let str = NSString(try JSONSerialize(obj, newIndent: newIndent, newLine: newLine, space: spacing))
+        return str.dataUsingEncoding(NSUTF8StringEncoding)!
+    }
+    
+    private class func JSONSerialize(object: AnyObject, newIndent: String, newLine: String, space: String, indentation: String = "") throws -> String
```

Comment:
object needs to be of type Any not AnyObject since this will not work for value types.